### PR TITLE
Calico: Mount /etc/resolv.conf so DNS works

### LINKED ISSUE #651: Calico doesn't resolve hostnames
Reported by a user on Twitter:

> @coreoslinux k8s setup guide https://coreos.com/kubernetes/docs/latest/deploy-master.html uses rkt for calico node - doesn’t resolve anything other than local host :/

https://twitter.com/alexouzounis/status/771392935204192256

> @coreoslinux after spending hours trying to realise that it’s the resolver, I got fed up and used @docker. Worked straight away

https://twitter.com/alexouzounis/status/771393213722685440

> @coreoslinux @docker issue is the etcd end points works only for local host. If you want to use rkt, one would need to mnt /etc/resolv.conf

https://twitter.com/alexouzounis/status/771393720239423489

### PR DESCRIPTION
**Pull request by tomdee**
Fixes #651

### CODE REVIEW

**Comment by robszumski** [comment]
I'd like to see this use the `RKT_OPTS` variable similar to https://github.com/coreos/coreos-kubernetes/pull/650 ([example](https://github.com/coreos/coreos-kubernetes/pull/650/files#diff-602bf92bbbb33b5e28e6b8ede6821f98R117)) so that we can make the wrapper look very similar to if you were just using the same flags on the actual kubelet. It makes it easier to understand whats going on, vs mixing in rkt flags with k8s flags.

**Comment by robszumski** [comment]
ping @tomdee does that sound ok?

**Comment by aaronlevy** [comment]
@robszumski I might be misunderstanding, but the RKT_OPTS is for passing options to the kubelet wrapper - these flag changes are for executing the calico container. Are you suggesting we may a calico wrapper script instead?

File: Documentation/deploy-master.md [hunk 1]
```diff
@@ -350,6 +350,8 @@ Environment=ETCD_ENDPOINTS=${ETCD_ENDPOINTS}
 ExecStart=/usr/bin/rkt run --inherit-env --stage1-from-dir=stage1-fly.aci \
 --volume=modules,kind=host,source=/lib/modules,readOnly=false \
 --mount=volume=modules,target=/lib/modules \
+--volume=dns,kind=host,source=/etc/resolv.conf,readOnly=false \
```
Comment (aaronlevy) [inline Documentation/deploy-master.md]: Why do you need write access to host resolv.conf?

**Comment by robszumski** [comment]
> @robszumski I might be misunderstanding, but the RKT_OPTS is for passing options to the kubelet wrapper - these flag changes are for executing the calico container. Are you suggesting we may a calico wrapper script instead?

Oh right. Disregard me.

**Comment by tomdee** [comment]
OK, changed to readonly=true, PTAL.

**Comment by tomdee** [comment]
I should give it another test - probably won't be today though.

**Comment by tomdee** [comment]
OK, this is now tested and is good to merge. @aaronlevy @robszumski

**Pull request merged**